src/gt4py/next/program_processors/runners/dace/gtir_sdfg.py

```diff
@@ -803,18 +755,18 @@ def visit_FunCall(
         elif cpm.is_applied_as_fieldop(node):
             return gtir_builtin_translators.translate_as_fieldop(node, sdfg, head_state, self)
         elif isinstance(node.fun, gtir.Lambda):
-            # Special handling of lambda arguments that can be lowered as symbolic expressions:
-            # when all the symrefs the lambda argument depends on are SDFG symbols, the argument
-            # can be passed to the nested SDFG by means of symbol mapping.
+            # Special handling of scalar arguments of a let-lambda that can be lowered
+            # as symbolic expressions: when all the GTIR-symbols the scalar expression
+            # depends on are dace symbols, the argument can be passed to the nested
+            # SDFG by means of symbol mapping.
             symbolic_args = {}
             for p, lambda_arg in zip(node.fun.params, node.args, strict=True):
                 if not isinstance(lambda_arg.type, ts.ScalarType):
                     continue
-                symrefs = symbol_ref_utils.collect_symbol_refs(
-                    lambda_arg, self.global_symbols.keys()
-                )
-                if all(symref in sdfg.symbols for symref in symrefs):
-                    symbolic_args[str(p.id)] = gtir_sdfg_utils.get_symbolic(lambda_arg)
+                # convert the scalar argument to a dace symbolic expression
```

Comment:
```suggestion
                # convert the scalar argument to a dace symbolic expression if all of its dependencies are symbols to.
```

I wonder what happens if you have the following case.
- You have symbol `a := symbolicParam1+ 3` and `b := a + 3`. 
- It is clear that you can handle `a` in that way because `param1` is a symbolic argument.
- However, for `b` you can do that too, however, after you have figured out that `a` is also a symbol.

This means that the order of processing is important, i.e. `a` must be processed before `b` is processed.
I am actually not sure if this can happen at all (and if I was able to express myself in an understandable way).

Reply:
You are right!